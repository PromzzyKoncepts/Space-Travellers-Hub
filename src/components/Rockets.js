/* eslint-disable camelcase */

import React from 'react';
import { useDispatch } from 'react-redux';
import { leaveRocket, reserveRocket } from '../Redux/RocketScreen/rocketscreen';

const Rockets = (item) => {
  const {
    flickr_images, rocket_name, description, id, reserved,
  } = item;
  const dispatch = useDispatch();
  return (
    <div className="card-container">
      <div>
        <img src={flickr_images} className="images" alt="rocket" />
      </div>
      <div>
        <h2>{rocket_name}</h2>
        <p>
          <span
            style={{
              backgroundColor: reserved ? '#007bff' : 'grey',
              color: reserved ? '#fff' : '#000',
              padding: reserved ? '5px 10px' : '0',
              marginRight: reserved ? '10px' : '0',
              borderRadius: reserved ? '5px' : '0',
            }}
          >
            {reserved && 'reserved'}
          </span>
          {description}
        </p>
        <button
          style={{ backgroundColor: reserved ? 'gray' : '#007bff' }}
          className="btn"
          type="button"
          onClick={
            reserved ? dispatch(leaveRocket(id)) : dispatch(reserveRocket(id))
          }
        >
          {reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
        </button>
      </div>
    </div>
  );
};

export default Rockets;
