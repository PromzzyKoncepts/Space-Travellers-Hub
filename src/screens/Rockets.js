/* eslint-disable camelcase */

import React from 'react';
// import { useDispatch } from 'react-redux';

const Rockets = (item) => {
  const { flickr_images, rocket_name, description } = item;
  console.log(flickr_images);
  return (
    <div className="card-container">
      <div>
        <img src={flickr_images} className="images" alt="rocket" />
      </div>
      <div>
        <h2>{rocket_name}</h2>
        <p>{description}</p>
        <button className="btn" type="button">Reserve Rocket</button>
      </div>
    </div>
  );
};

export default Rockets;
