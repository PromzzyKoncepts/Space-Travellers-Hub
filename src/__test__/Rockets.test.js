import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../Redux/ConfigureStore";
import Rockets from "../components/Rockets";

it("renders Component correctly", () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
