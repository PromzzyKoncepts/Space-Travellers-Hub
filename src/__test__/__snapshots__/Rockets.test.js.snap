// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component correctly 1`] = `
<div
  className="card-container"
>
  <div>
    <img
      alt="rocket"
      className="images"
    />
  </div>
  <div>
    <h2 />
    <p>
      <span
        style={
          Object {
            "backgroundColor": "grey",
            "borderRadius": "0",
            "color": "#000",
            "marginRight": "0",
            "padding": "0",
          }
        }
      />
    </p>
    <button
      className="btn"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#007bff",
        }
      }
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
